frontend/src/app/modules/header/header-item/header-item.component.ts

```diff
@@ -30,7 +30,14 @@ export class HeaderItemComponent extends BaseComponent implements OnInit {
     }
 
     ngOnInit(): void {
-        this.currentUser = this.userService.getUserFromStorage();
+        this.userService.userChangedEvent$.pipe(this.untilThis).subscribe({
+            next: (user) => {
+                if (user) {
+                    this.currentUser = user;
+                }
+            },
```

Comment:
what about error in this case ?

Reply:
I can't see how the error could appear there. If you mean "What if user is `undefined`" -  seems like the header component was designed to manage this edge case. Furthermore, the header shouldn't be reachable, if user is undefined (logged out), doesn't it?